mesonbuild/modules/_qt.py

```diff
@@ -648,3 +733,431 @@ def compile_translations(self, state: 'ModuleState', args: T.Tuple, kwargs: 'Com
+        #match DEPENDENCIES argument from CMake, but dependencies keyword is already taken
+        KwargInfo('depends_imports', ContainerTypeInfo(list, (str)), default=[]),
+        KwargInfo('designer_supported', bool, default=False),
+
+        #for type registration, same arguments as moc
+        KwargInfo('moc_headers', ContainerTypeInfo(list, (File, str, build.CustomTarget)), listify=True, default=[]),
+        KwargInfo('moc_sources', ContainerTypeInfo(list, (File, str, build.CustomTarget)), listify=True, default=[]),
+        KwargInfo('include_directories', ContainerTypeInfo(list, (build.IncludeDirs, str)), listify=True, default=[]),
+        KwargInfo('namespace', str, default=''),
+        KwargInfo('typeinfo', str, default=''),
+
+        KwargInfo('moc_extra_args', ContainerTypeInfo(list, str), listify=True, default=[]),
+        KwargInfo('qmlcachegen_extra_args', ContainerTypeInfo(list, str), listify=True, default=[]),
+        KwargInfo('qmltyperegistrar_extra_args', ContainerTypeInfo(list, str), listify=True, default=[]),
+
+        KwargInfo('generate_qmldir', bool, default=True),
+        KwargInfo('generate_qmltype', bool, default=True),
+        KwargInfo('cachegen', bool, default=True),
+
+        KwargInfo('dependencies', ContainerTypeInfo(list, (Dependency, ExternalLibrary)), listify=True, default=[]),
+        INSTALL_DIR_KW,
+        INSTALL_KW,
+        KwargInfo('method', str, default='auto'),
+        KwargInfo('preserve_paths', bool, default=False),
+    )
+    def qml_module(self, state: ModuleState, args: T.Tuple[str, str, str], kwargs: 'QmlModuleKwArgs') -> ModuleReturnValue:
+
+        self._detect_tools(state, kwargs['method'])
+        if not self._support_qml_module:
+            raise MesonException('qt.qml_module is not suppported for this version of Qt')
+
+        #Major.Minor(.Patch)
+        version_re = re.compile('^(\\d+)\\.(\\d+)(\\.(\\d+))?$')
+        output: T.List[T.Union[build.CustomTarget, build.GeneratedList]] = []
+
+        target_name: str = args[0]
+        module_name: str = args[1]
+        module_version: str = args[2]
+
+        module_version_match = version_re.match(module_version)
+        if not module_version_match:
+            raise MesonException(f'qml module version should be in the form Major.Minor, got {module_version}')
+        module_version_major: str = module_version_match.group(1)
+        module_version_minor: str = module_version_match.group(2)
+        #qt ignores .patch version
+
+        module_prefix_list: T.List[str] = module_name.split('.')
+        module_prefix: str = os.path.join(*module_prefix_list)
+        module_prefix_full: str = os.path.join(*(kwargs['resources_prefix'].split('/') + module_prefix_list))
+
+        qrc_resouces: T.List[T.Union[FileOrString, build.GeneratedTypes]] = []
+        all_qml: T.Sequence[T.Union[FileOrString, build.GeneratedTypes]] = list(kwargs['qml_sources']) + list(kwargs['qml_singletons']) + list(kwargs['qml_internals'])
```

Comment:
Aren't these already lists? Why cast them again to list?

Reply:
mypy fails otherwise

```
Unsupported left operand type for + ("Sequence[File | str | CustomTarget | CustomTargetIndex | GeneratedList]")
```

Reply:
Maybe the original definition of kwargs should annotate these as T.List instead of the generic T.Sequence, if we need them to be a List type?

Reply:
I changed it for a T.List, this seems to be enough